src/UhdmAst.cpp

```diff
@@ -486,6 +486,50 @@ AstNodeDType* getDType(vpiHandle obj_h, UhdmShared& shared) {
     return dtype;
 }
 
+AstNode* process_assignment(vpiHandle obj_h, UhdmShared& shared) {
+    AstNode* lvaluep = nullptr;
+    AstNode* rvaluep = nullptr;
+    const unsigned int objectType = vpi_get(vpiType, obj_h);
+
+    // Right
+    visit_one_to_one({vpiRhs}, obj_h, shared, [&](AstNode* childp) { rvaluep = childp; });
+
+    // Left
+    visit_one_to_one({vpiLhs}, obj_h, shared, [&](AstNode* childp) { lvaluep = childp; });
+
+    if (rvaluep != nullptr && rvaluep->type() == AstType::en::atFOpen) {
+        // Not really an assignpment, AstFOpen node takes care of the lhs
+        return rvaluep;
+    }
+    if (lvaluep && rvaluep) {
+        if (objectType == vpiAssignment) {
+            auto blocking = vpi_get(vpiBlocking, obj_h);
+            if (blocking) {
+                return new AstAssign(new FileLine("uhdm"), lvaluep, rvaluep);
+            } else {
+                return new AstAssignDly(new FileLine("uhdm"), lvaluep, rvaluep);
+            }
+        } else {
+            AstNode* assignp;
+            if (lvaluep->type() == AstType::en::atVar) {
```

Comment:
I think the standard Verilator idiom for this is `if (auto* varp = VN_CAST(lvaluep, Var))` (or `VN_IS` if you don't need the casted pointer). See:
https://github.com/verilator/verilator/blob/9650aefa4234808df884af3b4845e102cc099274/src/V3Const.cpp#L100